tests/src/JIT/Directed/Misc/Concat/ConcatTest.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+internal class ConcatTest
+{
+    static string strA = "A";
+    static string strB = "B";
+    static string strC = "C";
+    static string strD = "D";
+    static string strE = "E";
+
+    static string strAB      = "AB";
+    static string strABC     = "ABC";
+    static string strABCD    = "ABCD";
+    static string strABCDE   = "ABCDE";
+    static string strABCDx2  = "ABCDABCD";
+
+    static int iReturn = 100;
+
+    static public int Main()
+    {
+        iReturn = 100;
+        try
+        {
+            string result;
+
+            result = string.Concat(strA, strB);
```

Comment:
> there were really only an handful of them elsewhere

Note that majority of the framework test cases live in corefx repo. We do not have, nor trying to have, a comprehensive API test coverage in CoreCLR repo.

> It looked to me that an older version of the C# compiler would possibly create a call to the vararg version automatically when 4 or more arguments were passed to Concat

The C# compiler would not ever call the vararg version. You would have to use the undocumented `__arglist` keyword.